python/pyarrow/_dataset.pyx

```diff
@@ -1819,6 +1824,28 @@ cdef class CsvFragmentScanOptions(FragmentScanOptions):
                                         self.read_options)
 
 
+cdef class CsvFileWriteOptions(FileWriteOptions):
```

Comment:
same nit on Csv vs CSV

Reply:
Unfortunately in the context of datasets (and only datasets) all other classes already use Csv.